### DISCUSSION TITLE
Leverage the WP CLI or Node CLI to automatically implement CSS frameworks like Bootstrap, Tailwind, etc.

### DISCUSSION THREAD

**Discussion by robruiz** [wprig/wprig#693] [Ideas]
Having an easy way to prep the framework for the usage of a variety preselected CSS frameworks would be extremely handy. It would also be a nice way to attract new users to the framework if they are already religious about a particular framework.

**Comment by LukaszJaro** [comment]
I take this back. First I thought this could be useful so I could use my favorite CSS framework with Gutenberg with blocks. However CSS frameworks don't work well and offer less value with blocks, especially since blocks already use flexbox and grid.

I do a lot of my design/functionality in the block editor, many templates can be created with blocks. Full-site editing will make this even more apparent where literally every template you can create in the editor.

You would have to somehow properly integrate these frameworks with block editor. Maybe wprig spoiled me with being super clean, minimal, little dependencies and organized and I'm missing the point but maybe if we can get some more thoughts about this and the benefits, Maybe it would be more beneficial for the rig to have optional CSS for different layouts, like there is optional templates?

Seems like with the Gutenberg project and modern CSS, CSS frameworks are losing ground. Bootstrap does have many nice js components and helper classes, but you would have to integrate the components somehow into blocks to be most user friendly.

WPRIG really shines for these reasons more then ever, its made to integrate well with the direction WordPress is going, not sure about the full-site editing/json file though, that maybe requires wprig 3.0.

**Comment by robruiz** [comment]
The idea behind this would be to have it optional. Not encouraged, but
supported. For people who have an existing code-base of specific
requirements that require the use of one. Of these libraries.
